arch/riscv/Kconfig

```diff
@@ -280,6 +281,10 @@ config ARCH_IRQ_VECTOR_TABLE_ALIGN
 config GEN_IRQ_VECTOR_TABLE
 	select RISCV_MTVEC_VECTORED_MODE if SOC_FAMILY_RISCV_PRIVILEGE
 
+config ARCH_HAS_SINGLE_THREAD_SUPPORT
```

Comment:
Why this is defined here? This is already defined into `arch/Kconfig` and it is supposed to be selected as
```
select ARCH_HAS_SINGLE_THREAD_SUPPORT if !SMP
```
at SoC / platform / ISA level.

Reply:
`arch/Kconfig` only defines ARCH_HAS_SINGLE_THREAD_SUPPORT.
https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/Kconfig#L543-L544

The quoting is perhaps `arch/arc/Kconfig`.

https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/arc/Kconfig#L34
or
https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/arc/Kconfig#L42

So, We also need to consider the SMP condition in riscv.

Reply:
I'm still confused why you have to re-define this here.

Reply:
This change intends to enable ARCH_HAS_SINGLE_THREAD_SUPPORT by default (excluding if "!SMP").
This configuration is necessary to allow setting MULTITHREADING=n.

Essentially, Enabling ARCH_HAS_SINGLE_THREAD_SUPPORT is necessary, and the issue is where to do it.

For arm64, select and enable SINGLE_THREAD_SUPPORT by the commonly used CPU_CORTEX_A configuration.
For RISCV, there is no such generic setting, so I override it.